pkg/builder/local_build_executor.go

```diff
@@ -104,6 +107,8 @@ func (be *localBuildExecutor) uploadDirectory(ctx context.Context, outputDirecto
 				Name:   name,
 				Target: target,
 			})
+		case filesystem.FileTypeCharacterDevice:
```

Comment:
Oh, wait. You do check for the file type here.

Question: What makes character devices so special in this case? What about block devices? Maybe it would be acceptable to simply skip any non-file/device/symlink? I can imagine that FIFOs, UNIX sockets, etc. inside of output directories are also problematic.

Reply:
Nothing really makes them special I suppose, they were just the only scope of this PR and I didn't want to just remove the failure state for `FileTypeOther` and replace it with a `continue`. If that turns out to be the more desirable solution, or if there's something else that should be done instead I'm more than happy to change it up.

Reply:
Skipping all `FileTypeOther` entries is fine in my opinion.